plugins/needs_restarting.py

```diff
@@ -40,6 +40,113 @@
 NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus',
                'dbus-broker', 'dbus-daemon']
 
+class DirectoryOption(object):
+    """ This class handles a directory containing file options
+        This class can sanitize dir path
+    """
+    def __init__(self, path=None):
+        self._path = self._parse(path)
+
+    def _parse(self, path):
+        if not isinstance(path, str):
+            return set(path)
+        strval = path.replace('\n', ' ')
+        strval = path.replace(',', ' ')
+        strval = { x for x in strval.split(' ') if x }
+        strval = { x for x in strval if self._is_dir(x) }
+        return strval
+
+    def _is_dir(self, path=None):
+        return os.path.isdir(path)
+
+    #def _is_empty(self):
+    #    if len(os.listdir(self._path) ) == 0:
+    #        return False
+    #    return True
+
+    @property
+    def path(self):
+        return self._path
+
+class FileOption(object):
+    """This class takes a list of files and sanitizes them
+    """
+    def __init__(self, path):
+        self._path = path
+        self._files = self._read_config_files_in_dir()
+
+    def _read_config_files_in_dir(self):
+        files = set()
+        for dir in (self._path):
+            for file in os.listdir(dir):
+                files.add(os.path.join(dir, file))
+        return self._sanitize(files)
+
+    def _sanitize(self, files):
+        sane = set()
+        for file in files:
+            if not self._is_conf(file):
+                continue
+            if self._is_dir(file):
+                continue
+            sane.add(file)
+        return sane
+
+    def _is_dir(self, filename):
+        return os.path.isdir(filename)
+
+    def _is_conf(self, filename, extension='.conf'):
+        return filename.endswith(extension)
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def files(self):
+        return self._files
+
+class Option(object):
+    """ This class handles options inside .conf file
+    """
+    def __init__(self, path):
+        self._d_opt = DirectoryOption(path)
+        self._path = self._d_opt.path
```

Comment:
The business code of the `DirectoryOption` class is roughly 3 LOCs :slightly_smiling_face: The rest is just boilerplate (glue). Please minimize the amount of boilerplate unless the infrastructure you're putting up is actually needed :slightly_smiling_face: